libs/beacon/src/beacon_manager.cpp

```diff
@@ -644,14 +644,15 @@ bool BeaconManager::Verify(Signature const &signature)
 /**
  * @brief verifies a signed message by the group signature, where all parameters are specified.
  */
-bool BeaconManager::Verify(std::string const &group_public_key, MessagePayload const &message,
-                           std::string const &signature)
+bool BeaconManager::Verify(byte_array::ConstByteArray const &group_public_key,
+                           MessagePayload const &            message,
+                           byte_array::ConstByteArray const &signature)
 {
   PublicKey tmp;
-  tmp.setStr(group_public_key);
+  tmp.setStr(std::string(group_public_key));
 
   Signature tmp2;
-  tmp2.setStr(signature);
+  tmp2.setStr(std::string(signature));
```

Comment:
```suggestion
  tmp2.setStr(static_cast<std::string?(signature));
```

Reply:
this is constructing a string, not casting it

Reply:
So does the above, but I am not too phased either way